### DISCUSSION TITLE
Is it possible to build on a modern system?

### DISCUSSION THREAD

**Discussion by dark-penguin** [DeaDBeeF-Player/deadbeef#3068] [General]
Would it be possible to build deadbeef on a modern system, e.g. at least Debian Stable (Bookworm)? I'm trying (following the build instructions), but I'm running into all kinds of problems - mostly stuff being too outdated to be in the repos (gtk2, libdispatch...)

Barring that, what's the actual build process used nowadays? I'd assume something involving building a static build with Docker, but I couldn't figure it out completely on my own.

(I want to contribute, but I have to test my changes first, and for that, I need to figure out how to build the project. I'd be willing to help update the docs so that other potential contributors would be able to figure it out quickly.)

**Comment by Oleksiy-Yakovenko** [comment]
Yes, it should be fine to build on any compatible system, as long as you are able to follow instructions from the [README](https://github.com/DeaDBeeF-Player/deadbeef/blob/master/README) file.

I can't answer specifically about Debian Bookworm, since I haven't tried that, but Deadbeef builds fine on Ubuntu LTS 22.04, and this I do pretty regularly.

Are there some specific issues that you've encountered?

**Comment by Oleksiy-Yakovenko** [comment]
I don't know whether Debian ships Clang/libdispatch/blocksRuntime, but you would need to get those installed in order to build Deadbeef. They are not optional. GTK2 is optional though. You can use GTK3 instead.

**Comment by dark-penguin** [comment]
Thanks! I'll try again, probably also on Ubuntu 22.04. Let's figure out the problems here, then I'll document all the issues I've encountered.

**Comment by dark-penguin** [comment]
I'm trying to build on Ubuntu 22.04 (in docker). I'm following the README, but I've also tried to install the deps listed in `tools/docker/18.04/Dockerfile-builder`.

`CC=clang CXX=clang++ ./configure` fails with:
```
./configure: line 26077: syntax error near unexpected token `ZIP,'
./configure: line 26077: `    PKG_CHECK_MODULES(ZIP, libzip, HAVE_ZIP=yes, HAVE_ZIP=no)'
```
I've tried installing everything I could think of that has anything to do with "zip", but that did not help.

**Reply by dark-penguin** [reply]
Oh, seems like I have to rerun `autogen.sh` after I install it =/

**Reply by dark-penguin** [reply]
But no, I still can't figure out which package am I missing for the "ZIP" check

**Reply by dark-penguin** [reply]
It was `pkg-config`.

**Reply by dark-penguin** [reply]
Is there a list of Debian/Ubuntu dependency package names for the plugins? I'm trying to figure them out one by one. Maybe a Dockerfile you use to build the release with all plugins?

**Reply by dark-penguin** [reply]
I've figured them out, next I'll try to figure out the Debian package building procedure.

**Comment by dark-penguin** [comment]
So, there's no `debian/` folder to build a package "the usual way", but I see some kind of Travis configuration that is probably used to build releases, so it should be able to build it. I've tried to reproduce the build manually and tried the Docker instructions. Both of the approaches eventually failed with:
```
pw.c:19:10: fatal error: 'spa/param/audio/format-utils.h' file not found
#include <spa/param/audio/format-utils.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Is this some more missing deps?

**Reply by dark-penguin** [reply]
I should mention that I've managed to build it "normally", so looks like all the deps are in place. But I can't find a way to create a package (or even find where did it put all the artifacts...)

**Reply by dark-penguin** [reply]
Actually, now that I can `make install` it, I can make a Debian package out of it. And I can see that `debian.sh` does some more fixes to it, so there are my hints about what else should be done.
